lib/src/dart3_suggestors/null_safety_prep/required_flux_props.dart

```diff
@@ -0,0 +1,228 @@
+    _result = await context.getResolvedUnit();
+    if (_result == null) {
+      throw Exception(
+          'Could not get resolved result for "${context.relativePath}"');
+    }
+    _result!.unit.accept(this);
+  }
+}
+
+String? _getNameOfVarOrFieldInScopeWithType(AstNode node, DartType? type) {
+  final inScopeVariableDetector = _InScopeVarDetector();
+  // Find top level vars
+  node.thisOrAncestorOfType<CompilationUnit>()?.accept(inScopeVariableDetector);
+  // Find vars declared in top-level fns (like `main()`)
+  node
+      .thisOrAncestorOfType<BlockFunctionBody>()
+      ?.visitChildren(inScopeVariableDetector);
+
+  final inScopeVarName = inScopeVariableDetector.found
+      .firstWhereOrNull((v) {
+        final maybeMatchingType = v.declaredElement?.type;
+        return maybeMatchingType?.element?.name == type?.element?.name;
+      })
+      ?.declaredElement
+      ?.name;
+
+  final componentScopePropDetector = _ComponentScopeFluxPropsDetector();
+  // Find actions/store in props of class components
+  node
+      .thisOrAncestorOfType<ClassDeclaration>()
+      ?.accept(componentScopePropDetector);
+  // Find actions/store in props of fn components
+  node
+      .thisOrAncestorOfType<MethodInvocation>()
+      ?.accept(componentScopePropDetector);
+
+  final inScopePropName =
+      componentScopePropDetector.found.firstWhereOrNull((el) {
+    final maybeMatchingType = componentScopePropDetector.getAccessorType(el);
+    return maybeMatchingType?.element?.name == type?.element?.name;
+  })?.name;
+
+  if (inScopeVarName != null && inScopePropName != null) {
+    // TODO: Do we need to handle this edge case with something better than returning null?
+    // No way to determine which should be used - the scoped variable or the field on props
+    // so return null to avoid setting the incorrect value on the consumer's code.
+    return null;
+  }
+
+  if (inScopePropName != null) {
+    return '${componentScopePropDetector.propsName}.${inScopePropName}';
+  }
+
+  return inScopeVarName;
+}
+
+bool _isFnComponentDeclaration(Expression? varInitializer) =>
+    varInitializer is MethodInvocation &&
+    varInitializer.methodName.name.startsWith('uiF');
+
+/// A visitor to detect in-scope store/actions variables (top-level and block function scopes)
+class _InScopeVarDetector extends RecursiveAstVisitor<void> {
+  final List<VariableDeclaration> found;
+
+  _InScopeVarDetector() : found = [];
+
+  @override
+  visitVariableDeclaration(VariableDeclaration node) {
+    // Don't visit function component declarations here since we visit them using the _ComponentScopeFluxPropsDetector
+    if (_isFnComponentDeclaration(node.initializer)) return;
+
+    if (node.declaredElement != null) {
+      found.add(node);
+    }
+  }
+}
+
+/// A visitor to detect store/actions values in a props class (supports both class and fn components)
+class _ComponentScopeFluxPropsDetector extends RecursiveAstVisitor<void> {
```

Comment:
#nit it doesn't look like this class is really being used as a visitor, since it only processes the nodes we directly pass to it.

We could make it a non-visitor and it would function just the same:
```diff
diff --git a/lib/src/dart3_suggestors/null_safety_prep/required_flux_props.dart b/lib/src/dart3_suggestors/null_safety_prep/required_flux_props.dart
index 9d84eb9..3617886 100644
--- a/lib/src/dart3_suggestors/null_safety_prep/required_flux_props.dart
+++ b/lib/src/dart3_suggestors/null_safety_prep/required_flux_props.dart
@@ -115,13 +115,11 @@ String? _getNameOfVarOrFieldInScopeWithType(AstNode node, DartType? type) {
 
   final componentScopePropDetector = _ComponentScopeFluxPropsDetector();
   // Find actions/store in props of class components
-  node
-      .thisOrAncestorOfType<ClassDeclaration>()
-      ?.accept(componentScopePropDetector);
+  componentScopePropDetector.handlePotentialClassComponent(
+      node.thisOrAncestorOfType<ClassDeclaration>());
   // Find actions/store in props of fn components
-  node
-      .thisOrAncestorOfType<MethodInvocation>()
-      ?.accept(componentScopePropDetector);
+  componentScopePropDetector.handlePotentialFunctionComponent(
+      node.thisOrAncestorOfType<MethodInvocation>());
 
   final inScopePropName =
       componentScopePropDetector.found.firstWhereOrNull((el) {
@@ -165,8 +163,9 @@ class _InScopeVarDetector extends RecursiveAstVisitor<void> {
 }
 
 /// A visitor to detect store/actions values in a props class (supports both class and fn components)
-class _ComponentScopeFluxPropsDetector extends RecursiveAstVisitor<void> {
+class _ComponentScopeFluxPropsDetector {
   final Map<PropertyAccessorElement, DartType> _foundWithMappedTypes;
+
   List<PropertyAccessorElement> get found =>
       _foundWithMappedTypes.keys.toList();
 
@@ -201,8 +200,8 @@ class _ComponentScopeFluxPropsDetector extends RecursiveAstVisitor<void> {
   }
 
   /// Visit function components
-  @override
-  void visitMethodInvocation(MethodInvocation node) {
+  void handlePotentialFunctionComponent(MethodInvocation? node) {
+    if (node == null) return;
     if (!_isFnComponentDeclaration(node)) return;
 
     final nodeType = node.staticType;
@@ -219,8 +218,8 @@ class _ComponentScopeFluxPropsDetector extends RecursiveAstVisitor<void> {
   }
 
   /// Visit composite (class) components
-  @override
-  void visitClassDeclaration(ClassDeclaration node) {
+  void handlePotentialClassComponent(ClassDeclaration? node) {
+    if (node == null) return;
     final elWithProps =
         node.declaredElement?.supertype?.typeArguments.singleOrNull?.element;
     _lookForFluxStoreAndActionsInPropsClass(elWithProps);
```

Reply:
Addressed in https://github.com/Workiva/over_react_codemod/pull/268/commits/619910d7e5bd4bf060862cac0d95a97ad8b2e327